docs/src/architecture/hardening.md

```diff
@@ -23,26 +23,28 @@ NixOS provides several mechanisms to customize kernel. The main methods are:
   ```
   ~/ghaf $ nix develop .#devShells.x86_64-linux.kernel-x86
   ...
-  [ghaf-kernel-devshell:~/ghaf/linux-6.6.2]$ cp ../modules/host/ghaf_host_hardened_baseline .config
-  [ghaf-kernel-devshell:~/ghaf/linux-6.6.2]$ make menuconfig
+  [ghaf-kernel-devshell:~/ghaf/linux-6.6.5]$ cp ../modules/host/ghaf_host_hardened_baseline .config
+  [ghaf-kernel-devshell:~/ghaf/linux-6.6.5]$ make menuconfig
   ...
-  [ghaf-kernel-devshell:~/ghaf/linux-6.6.2]$ make -j$(nproc)
+  [ghaf-kernel-devshell:~/ghaf/linux-6.6.5]$ make -j$(nproc)
   ...
   Kernel: arch/x86/boot/bzImage
   ```
 * Boot the built kernel with QEMU
   ```
-  [ghaf-kernel-devshell:~/ghaf/linux-6.6.2]$ qemu-system-x86_64 -kernel arch/x86/boot/bzImage
+  [ghaf-kernel-devshell:~/ghaf/linux-6.6.5]$ qemu-system-x86_64 -kernel arch/x86/boot/bzImage
   ```
 * [validating with kernel hardening checker](https://github.com/a13xp0p0v/kernel-hardening-checker)
   ```
-  [ghaf-kernel-devshell:~/ghaf/linux-6.6.2]$ kernel-hardening-checker -c ../modules/host/ghaf_host_hardened_baseline
+  [ghaf-kernel-devshell:~/ghaf/linux-6.6.5]$ kernel-hardening-checker -c ../modules/host/ghaf_host_hardened_baseline
   [+] Kconfig file to check: ../modules/host/ghaf_host_hardened_baseline
   [+] Detected microarchitecture: X86_64
   [+] Detected kernel version: 6.6
-  [+] Detected compiler: GCC 120200
+  [+] Detected compiler: GCC 120300
   ...
-  [+] Config check is finished: 'OK' - 103 / 'FAIL' - 84
+  [+] Config check is finished: 'OK' - 188 / 'FAIL' - 5
+  Note:
+  'FAIL' - 5 cannot be fixed
```

Comment:
We must elaborate this to details:
```
[ghaf-kernel-devshell:~/ghaf/linux-6.6.5]$ kernel-hardening-checker -c ../modules/host/ghaf_host_hardened_baseline|grep 'FAIL: '
CONFIG_CFI_CLANG                        |kconfig|     y      |   kspp   | self_protection  | FAIL: is not found
CONFIG_CFI_PERMISSIVE                   |kconfig| is not set |   kspp   | self_protection  | FAIL: CONFIG_CFI_CLANG is not "y"
CONFIG_MODULES                          |kconfig| is not set |   kspp   |cut_attack_surface| FAIL: "y"
CONFIG_FB                               |kconfig| is not set |maintainer|cut_attack_surface| FAIL: "y"
CONFIG_VT                               |kconfig| is not set |maintainer|cut_attack_surface| FAIL: "y"
``` 
Please replace the lines https://github.com/tiiuae/ghaf/pull/415/files#diff-68871c88d917d2e369f25152d9444d2c1c6b57dfd55a912dbea01d6b2f9e3829R46-R47
with above snippet.

We can do more work for these but we can scope it out of this PR.
E.g. `CONFIG_[FB|VT]` we could disable on the baseline config but we must enable for `-debug`-targets. That could be done using a nix module that brings these debug essential configs with `structuredExtraConfig`
Similarly, `CONFIG_MODULES n` would cut attack surface significantly but we'd have to check carefully if the kernel can still be compiled. When I tested this PR, `lsmod` shows no loaded modules but there still are some options configured as modules.

Reply:
Fixed.